src/components/Cart/User/SignUp.vue

```diff
@@ -61,33 +63,45 @@ export default defineComponent({
     ErrorMessage
   },
   setup() {
+    const store = useStore();
     const router = useRouter();
     const { t } = useI18n();
+    const fbError: Ref<string> = ref("");
     const onSubmit = (values: any) => {
       auth
         .createUserWithEmailAndPassword(values.email, values.password)
-        .then(user => {
-          console.log(user.user);
-        });
+        .then(() => {
+          const user = {
+            name: values.name,
+            email: values.email
+          };
+          store.commit("user/setUpUser", user);
+        })
+        .then(() => router.push("/"))
+        .catch(err => (fbError.value = err.message));
     };
     const goBack = () => {
```

Comment:
As above. `goBack` sholud be stored in hook. Otherwise you are doubling your code.